lib/combine_pdf/renderer.rb

```diff
@@ -107,7 +107,7 @@ def format_hash_to_pdf(object)
         end
       end
       # remove extra page references.
-      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page
+      object[:Contents].delete(is_reference_only: true, referenced_object: { indirect_reference_id: 0, raw_stream_content: '' }) if object[:Type] == :Page && object[:Contents]
```

Comment:
This line is used to remove "empty" edits to PDF pages. This means that pages were expected to have data added to them but where eventually left alone (or the data added was another PDF page instead of a text object).

It seems that in your case you have a simple PDF file that wasn't edited by CombinePDF (i.e., you didn't add page numbers or used the stumping feature).

Such cases have `Contents` as either `nil` (if there's an error in the PDF file) or a `Hash` (as actual page), they don't require us to remove "empty" edits. Hence, the `if` statement can safely ignore instances where the type of the `Contents` isn't an array.

The array `#delete` function will delete a `Hash` that is equal to the hash passed along, meaning it will remove "empty" edit placeholders.